### PR TITLE
Update harvester charts to Rancher v2.7

### PR DESCRIPTION
**Pull request by chrisho**
This PR adds below charts to Rancher v2.7:
1. harvester-cloud-provider
2. harvester-csi-driver

**Related:**
https://github.com/rancher/rancher/issues/38975
https://github.com/rancher/rancher/issues/38976

### CODE REVIEW

**Comment by eliyamlevy** [comment]
Build is failing on this PR, make sure to run make validate before pushing to make sure that everything is passing.

File: packages/harvester/harvester-csi-driver/generated-changes/patch/Chart.yaml.patch [hunk 1]
```diff
@@ -10,7 +10,8 @@
    catalog.cattle.io/os: linux
 -  catalog.cattle.io/permits-os: linux
 +  catalog.cattle.io/permits-os: linux,windows
-   catalog.cattle.io/rancher-version: '>= 2.6.1-0 < 2.7.0-0'
+-  catalog.cattle.io/rancher-version: '>= 2.6.1-0 < 2.7.0-0'
++  catalog.cattle.io/rancher-version: '>= 2.6.1-0 < 2.8.0-0'
```
Comment (aiyengar2) [inline packages/harvester/harvester-csi-driver/generated-changes/patch/Chart.yaml.patch:14]: should be >= 2.7.0-0 < 2.8.0-0, a 101 chart would never be tested in 2.6

**Review by aiyengar2 — CHANGES_REQUESTED** [review]
commits need to be split between hand-made changes (packages, release.yaml) in first commit and generated changes (`make charts`) in second commit.

File: packages/harvester/harvester-csi-driver/generated-changes/patch/Chart.yaml.patch (hunk 1 above)
Comment (chrisho) [reply packages/harvester/harvester-csi-driver/generated-changes/patch/Chart.yaml.patch]: > should be >= 2.7.0-0 < 2.8.0-0, a 101 chart would never be tested in 2.6

How to be compatible going forward >= 2.6.0-0 < 2.7.0-0

**Review by eliyamlevy — APPROVED** [review]
LGTM @aiyengar2

**Review by aiyengar2 — APPROVED** [review]

**Comment by prachidamle** [comment]
@chrisho merging this now that the PR is approved.

**Pull request merged**